moveit_core/robot_model/src/floating_joint_model.cpp

```diff
@@ -136,8 +136,9 @@ void FloatingJointModel::interpolate(const double* from, const double* to, const
   state[1] = from[1] + (to[1] - from[1]) * t;
   state[2] = from[2] + (to[2] - from[2]) * t;
 
-  double dq = fabs(from[3] * to[3] + from[4] * to[4] + from[5] * to[5] + from[6] * to[6]);
-  double theta = (dq + std::numeric_limits<double>::epsilon() >= 1.0) ? 0.0 : acos(dq);
+  double dq = from[3] * to[3] + from[4] * to[4] + from[5] * to[5] + from[6] * to[6];
+  double theta = (fabs(dq) + std::numeric_limits<double>::epsilon() >= 1.0) ? 0.0 : acos(fabs(dq));
```

Comment:
Isn't there a factor of 2 missing here?

[This website](https://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/slerp/index.htm) gives this formula, note the 2:

`cos(theta/2) = qa.w*qb.w + qa.x*qb.x + qa.y*qb.y+ qa.z*qb.z`

Reply:
I'm not sure if it's really missing, the implementation matches the example code on the website you linked. `theta` could probably be renamed to `halfTheta`, though.

Reply:
The factor 2 comes from converting between axis-angle and quaternion. It's because quaternions are weird: you don't just multiply them with the thing you want to transform (though that's how most linalg libraries typically have their API set up) you pre-multiply with the quaternion, then post-multiply with the inverse quaternion.

Also, I don't see how the test could possible be passing if I was missing something as big as a factor 2 in there.